version/version.go

```diff
@@ -1,4 +1,51 @@
 package version
 
+import (
+	"strings"
+
+	"github.com/blang/semver"
+	"github.com/google/renameio"
+)
+
 // Version is the version of the build.
 const Version = "1.14.2-dev"
+
+// WriteVersionFile writes the version information to a given file
+// file is the location of the old version file
+// gitCommit is the current git commit version. It will be added to the file
+// to aid in debugging, but will not be used to compare versions
+func WriteVersionFile(file, gitCommit string) error {
+	current, err := parseVersionConstant(Version, gitCommit)
+	// Sanity check-this should never happen
+	if err != nil {
+		return err
+	}
+	json, err := current.MarshalJSON()
+	// Sanity check-this should never happen
+	if err != nil {
+		return err
+	}
+
+	return renameio.WriteFile(file, json, 0644)
```

Comment:
related to my previous comment.  Is it supposed to be on persistent storage?

If we write to non persistent storage, we can probably drop the additional dependency as the `fsync` doesn't have any effect on tmpfs.  We can replace it with a simpler write to a ioutil.TempFile() + `os.Rename()`

Reply:
@giuseppe it is supposed to persist, as it's used to check after a reboot whether the storage should be wiped.

Reply:
I went for /var/lib/crio/version